osf/models/node.py

```diff
@@ -445,6 +447,23 @@ def linked_nodes(self):
         ).select_related('child').values_list('child', flat=True)
         return self._nodes.filter(pk__in=child_pks)
 
+    @property
+    def root_folder(self):
+        """
+        Returns AbstractNode's root_folder. Either an OsfStorageFolder or a TrashedFolder type.
+        """
+        from osf.models.files import BaseFileNode
+
+        try:
+            return BaseFileNode.objects.get(
+                target_object_id=self.id,
+                target_content_type_id=ContentType.objects.get_for_model(AbstractNode).id,
+                name='',
+                is_root=True
+            )
+        except BaseFileNode.DoesNotExist:
+            return None
```

Comment:
Defining a root_folder property (mirrors the Preprint.root_folder property). Returns folder even if has been deleted.